recipe-server/client/control/tests/state/actions/test_selectors.js

```diff
@@ -0,0 +1,37 @@
+import { fromJS } from 'immutable';
+
+import { getAction } from 'control/state/actions/selectors';
+
+import {
+  ACTION,
+} from '.';
+
+import {
+  INITIAL_STATE,
+} from '..';
+
+
+describe('getAction', () => {
+  const STATE = {
+    ...INITIAL_STATE,
+    newState: {
+      ...INITIAL_STATE.newState,
+      actions: {
+        ...INITIAL_STATE.newState.actions,
+        items: INITIAL_STATE.newState.actions.items.set(ACTION.id, fromJS(ACTION)),
```

Comment:
Making the whole tree immutable is a bit janky with the reducers from redux-form etc.